src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveredSource.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+
+#nullable disable
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+/// <summary>
+/// Describes a source file that was discovered, or that was requested to be discovered.
+/// </summary>
+public class DiscoveredSource
+{
+    [DataMember]
+    public string Source { get; set; }
```

Comment:
I know we need some setting for serialization but adding a private parameterless ctor and making the property setter private helps with flowing nullability in the codebase. Because we don't expect `Source` to be `null` but we will have to handle null everywhere this type is used in the current form.